### PR TITLE
i#3579 AVX-512 support: Add missing K type to disassembler.

### LINKED ISSUE #3579: TYPE_K_REG missing in disassembler.
When pulling in binutils tests I ran into the disassembler failing to print the new AVX-512 kmov instructions because TYPE_K_REG is missing from opnd_disassemble_noimplicit.

xref #3574

### PR DESCRIPTION
**Pull request by hgreving2304**
Add missing TYPE_K_REG to operand disassembler.

Fixes #3579
Issue: #1312

### CODE REVIEW

File: core/arch/x86/disassemble.c [hunk 1]
```diff
@@ -195,6 +195,7 @@ opnd_disassemble_noimplicit(char *buf, size_t bufsz, size_t *sofar INOUT,
     case TYPE_FLOATMEM:
```
Comment (derekbruening) [inline core/arch/x86/disassemble.c]: Please add a test to ir_x86.c where we have some disassembly output string comparisons.

Comment (hgreving2304) [reply core/arch/x86/disassemble.c]: I am adding an entire suite of tests right now along with #3574, these opcodes here will be included. It will take a little bit as I am going through the failures one by one. Re-open if necessary.

**Comment by AssadHashmi** [comment]
run arm tests

File: core/arch/x86/disassemble.c (hunk 1 above)
Comment (derekbruening) [reply core/arch/x86/disassemble.c]: OK, though we like to have corner cases or past bugs included in our own tests, which includes this.

Comment (hgreving2304) [reply core/arch/x86/disassemble.c]: I have added a small dissembler test to ir_x86.

**Comment by hgreving2304** [comment]
run arm tests
